lib/begin.go

```diff
@@ -188,6 +189,22 @@ func (a *ACBuild) writeEmptyManifest() error {
 		return err
 	}
 
+	archvalue := runtime.GOARCH
+	if runtime.GOOS == "linux" && archvalue == "arm" {
+		archvalue = "armv7l"
+		walker := func(path string, info os.FileInfo, err error) error {
```

Comment:
Oh, it's just the usual way to detect local endianness - put a known multibyte value into memory, and then see how the compiler decided to actually store it.  In this case it's seeing if 0x01020304 was stored with 0x01 or 0x04 first.